src/app/components/AddSource.svelte

```diff
@@ -0,0 +1,183 @@
+<script lang="ts">
+  import worker from "../spawn-worker";
+  import Button from "./Button.svelte";
+  import SplitButton from "./SplitButton.svelte";
+  import ManualEntry from "./ManualEntry.svelte";
+  const UPLOAD_INPUT_ID = "upload-input";
+
+  const leftText = "UPLOAD";
+  const rightText = "DIRECT ENTRY";
+  export let addSource: boolean = false;
+  let manualEntry: boolean = false;
+  let onDraggedOver = false;
+  let fileList: any = [];
+
+  function fileFromDataTransferItem(items: DataTransferItemList): File[] {
+    const fileList: File[] = [];
+    for (let item of items) {
+      // If dropped items aren't files, reject them
+      if (item.kind === "file") {
+        const file = item.getAsFile();
+        if (file !== null) {
+          fileList.push(file);
+        }
+      }
+    }
+    return fileList;
+  }
+
+  const handleDrop = async (event: DragEvent) => {
```

Comment:
```suggestion
  const handleDrop = async (event: DragEvent) => {
```
```suggestion
  const handleDrop = (event: DragEvent) => {
```
`Async` can be removed here if there is no `await` in the code